/**:
  ros__parameters:
    filter_chain_parameter_name: 'filter'

    filter:
      filter1:
        name: inpaint
        type: gridMapCv/InpaintFilter
        params:
          input_layer: elevation
          output_layer: elevation_inpainted
          radius: 0.11

      filter2:
        name: mean_in_radius
        type: gridMapFilters/MeanInRadiusFilter
        params:
          input_layer: elevation_inpainted
          output_layer: elevation_smooth
          radius: 0.11
          
      filter3:
        name: surface_normals
        type: gridMapFilters/NormalVectorsFilter
        params:
          input_layer: elevation_inpainted
          output_layers_prefix: normal_vectors_
          radius: 0.11
          normal_vector_positive_axis: z

      filter4:  # Compute slope from surface normal.
        name: slope
        type: gridMapFilters/MathExpressionFilter
        params:
          output_layer: slope
          expression: acos(normal_vectors_z)

      filter5: # Compute roughness as absolute difference from map to smoothened map.
        name: roughness
        type: gridMapFilters/MathExpressionFilter
        params:
          output_layer: roughness
          expression: abs(elevation_inpainted - elevation_smooth)

      # filter6: # Compute traversability as normalized weighted sum of slope and roughness.
      #   name: traversability
      #   type: gridMapFilters/MathExpressionFilter
      #   params:
      #     output_layer: traversability
      #     expression: 0.5 * (1.0 - (slope / 0.4)) + 0.5 * (1.0 - (roughness / 0.1))

      # filter7:  # Set lower threshold on traversability.
      #   name: traversability_lower_threshold
      #   type: gridMapFilters/ThresholdFilter
      #   params:
      #     layer: traversability
      #     lower_threshold: 0.0
      #     set_to: 0.0

      # filter8:  # Set upper threshold on traversability.
      #   name: traversability_upper_threshold
      #   type: gridMapFilters/ThresholdFilter
      #   params:
      #     layer: traversability
      #     upper_threshold: 1.0
      #     set_to: 1.0 # Other uses: .nan, .inf